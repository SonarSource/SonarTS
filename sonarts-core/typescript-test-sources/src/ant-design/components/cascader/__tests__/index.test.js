import React from 'react';
import { render, mount } from 'enzyme';
import KeyCode from 'rc-util/lib/KeyCode';
import Cascader from '..';

const options = [{
  value: 'zhejiang',
  label: 'Zhejiang',
  children: [{
    value: 'hangzhou',
    label: 'Hangzhou',
    children: [{
      value: 'xihu',
      label: 'West Lake',
    }],
  }],
}, {
  value: 'jiangsu',
  label: 'Jiangsu',
  children: [{
    value: 'nanjing',
    label: 'Nanjing',
    children: [{
      value: 'zhonghuamen',
      label: 'Zhong Hua Men',
    }],
  }],
}];

describe('Cascader', () => {
  it('popup correctly when panel is hidden', () => {
    const wrapper = mount(
      <Cascader options={options} />
    );
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
  });

  it('popup correctly when panel is open', () => {
    const wrapper = mount(
      <Cascader options={options} />
    );
    wrapper.find('input').simulate('click');
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
  });

  it('popup correctly with defaultValue', () => {
    const wrapper = mount(
      <Cascader options={options} defaultValue={['zhejiang', 'hangzhou']} />
    );
    wrapper.find('input').simulate('click');
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
  });

  it('can be selected', () => {
    const wrapper = mount(<Cascader options={options} />);
    wrapper.find('input').simulate('click');
    let popupWrapper = mount(wrapper.find('Trigger').node.getComponent());
    popupWrapper.find('.ant-cascader-menu').at(0).find('.ant-cascader-menu-item').at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
    popupWrapper = mount(wrapper.find('Trigger').node.getComponent());
    popupWrapper.find('.ant-cascader-menu').at(1).find('.ant-cascader-menu-item').at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
    popupWrapper = mount(wrapper.find('Trigger').node.getComponent());
    popupWrapper.find('.ant-cascader-menu').at(2).find('.ant-cascader-menu-item').at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').node.getComponent())).toMatchSnapshot();
  });

  it('backspace should work with `Cascader[showSearch]`', () => {
    const wrapper = mount(<Cascader options={options} showSearch />);
    wrapper.find('input').simulate('change', { target: { value: '123' } });
    expect(wrapper.state('inputValue')).toBe('123');
    wrapper.find('input').simulate('keydown', { keyCode: KeyCode.BACKSPACE });
    // Simulate onKeyDown will not trigger onChange by default, so the value is still '123'
    expect(wrapper.state('inputValue')).toBe('123');
  });
});
