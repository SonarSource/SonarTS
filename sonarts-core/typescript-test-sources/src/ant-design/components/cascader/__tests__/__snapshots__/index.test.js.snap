// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader can be selected 1`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-placement-bottomLeft "
  >
    <div>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
        </li>
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Hangzhou"
        >
          Hangzhou
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader can be selected 2`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-placement-bottomLeft "
  >
    <div>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
        </li>
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Hangzhou"
        >
          Hangzhou
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item"
          title="West Lake"
        >
          West Lake
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader can be selected 3`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-placement-bottomLeft  ant-cascader-menus-hidden"
  >
    <div>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
        </li>
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Hangzhou"
        >
          Hangzhou
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-active"
          title="West Lake"
        >
          West Lake
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader popup correctly when panel is hidden 1`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-empty ant-cascader-menus-placement-bottomLeft  ant-cascader-menus-hidden"
  >
    <div />
  </div>
</div>
`;

exports[`Cascader popup correctly when panel is open 1`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-placement-bottomLeft "
  >
    <div>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Zhejiang"
        >
          Zhejiang
        </li>
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader popup correctly with defaultValue 1`] = `
<div>
  <div
    class="ant-cascader-menus  ant-cascader-menus-placement-bottomLeft "
  >
    <div>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
        </li>
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item ant-cascader-menu-item-expand ant-cascader-menu-item-active"
          title="Hangzhou"
        >
          Hangzhou
        </li>
      </ul>
      <ul
        class="ant-cascader-menu"
      >
        <li
          class="ant-cascader-menu-item"
          title="West Lake"
        >
          West Lake
        </li>
      </ul>
    </div>
  </div>
</div>
`;
